### ISSUE TITLE
[CI] Java tests flaky segfault due to spdlog 

### ISSUE THREAD

**Issue by simon-mo** [ray-project/ray#13741]
Java unit tests fails pretty often due to a segfault:
```
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x00007fa59cf8e8d8, pid=19042, tid=0x00007fa59f457700
#
# JRE version: OpenJDK Runtime Environment (8.0_242-b08) (build 1.8.0_242-b08)
# Java VM: OpenJDK 64-Bit Server VM (25.242-b08 mixed mode linux-amd64 compressed oops)
# Problematic frame:
# C  [libcore_worker_library_java.so+0x6a78d8]  spdlog::details::registry::initialize_logger(std::shared_ptr<spdlog::logger>)+0x58
#
# Failed to write core dump. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again
#
# If you would like to submit a bug report, please visit:
#   http://bugreport.java.com/bugreport/crash.jsp
#
```
Here are the full Travis links:
https://travis-ci.com/github/ray-project/ray/jobs/474924913
https://travis-ci.com/github/ray-project/ray/jobs/475366727
https://travis-ci.com/github/ray-project/ray/jobs/476695007
https://travis-ci.com/github/ray-project/ray/jobs/477344700

**Comment by ashione** [comment]
I'll try to find why it's flakytest.

**Comment by raulchen** [comment]
This should be fixed by #13762
